### PR TITLE
WIP: Admin lockout fix

### PR DESCRIPTION
**Pull request by AstridKery**
This MR adds an admin timeout. It's currently set to 3 minutes to make it easier for reviewers to test it, but we'll probably want it to be more like an hour or two. @peridoteagle this is ready for review but we'll need to update the timeout before merging.

This MR borrows from an earlier one of @schittarath3-rti to update the AdminProgress table each time an admin with the lock preforms an action. The change is that if another admin enters the coding page, SMART merely checks if the first admin's last action was longer than a certain cutoff. If it is, the old admin's progress is replaced by the new admin. 

If the old admin still has the page loaded and tries to complete an action on the skew, recycle bin, or skipped pages, they won't work, and instead a message will display letting them know they lost access.

Caveat: currently when the second admin is given access, they have to refresh twice to get the unlocked page. This is because I think the enter_coding_page function is asynchronous and finishes after the initial refresh found that they were locked out. I don't think this is a problem so long as people know they might want to refresh again?

### CODE REVIEW

File: backend/django/core/utils/utils_annotate.py [hunk 1]
```diff
@@ -280,3 +280,17 @@ def cache_embeddings(project_pk, embeddings):
 
 def get_embeddings(project_pk):
     return cache.get(project_pk)
+
+
+def leave_admin_page(profile, project):
```
[the remark this answers is not in the document]
Comment (AstridKery) [reply backend/django/core/utils/utils_annotate.py]: Oh good point! I think I copied this over from Souliya's MR but didn't use it in the way he did. I just removed it.

File: backend/django/core/views/api_annotate.py [hunk 2]
```diff
@@ -420,12 +439,31 @@ def enter_coding_page(request, project_pk):
...
     if project_extras.proj_permission_level(project, profile) > 1:
         if AdminProgress.objects.filter(project=project).count() == 0:
             AdminProgress.objects.create(
                 project=project, profile=profile, timestamp=timezone.now()
             )
+        else:
+            # figure out the time from the last time the other admin used the page
+            previous_admin_progress = AdminProgress.objects.filter(project=project)
+            if previous_admin_progress[0].profile == profile:
+                print("This admin currently has access")
```
[the remark this answers is not in the document]
Comment (AstridKery) [reply backend/django/core/views/api_annotate.py]: They were here for debugging. I removed them

File: frontend/src/components/AdminTable/index.jsx [hunk 3]
```diff
@@ -77,6 +77,13 @@ class AdminTable extends React.Component {
         }
         page_sizes.push(admin_data.length);
 
+        if (message.length > 0){
+            let message_new = message[0];
+            if (message_new.includes("ERROR")){
+                return (<div>{message_new}</div>);
```
[the remark this answers is not in the document]
Comment (AstridKery) [reply frontend/src/components/AdminTable/index.jsx:83]: Good idea! I just updated the lockout message to add that in.

**Comment by AstridKery** [comment]
Thanks @dsteedRTI ! I think I've addressed your comments above

**Comment by dsteedRTI** [comment]
Looks good!

**Pull request merged**